models/stg_facebook_ads__basic_ad_actions.sql

```diff
@@ -0,0 +1,50 @@
+
+with base as (
+
+    select * 
+    from {{ ref('stg_facebook_ads__basic_ad_actions_tmp') }}
+),
+
+fields as (
+
+    select
+        {{
+            fivetran_utils.fill_staging_columns(
+                source_columns=adapter.get_columns_in_relation(ref('stg_facebook_ads__basic_ad_actions_tmp')),
+                staging_columns=get_basic_ad_actions_columns()
+            )
+        }}
+        {{ fivetran_utils.source_relation(
+            union_schema_variable='facebook_ads_union_schemas', 
+            union_database_variable='facebook_ads_union_databases') 
+        }}
+    from base
+),
+
+final as (
+    
+    select 
+        source_relation, 
+        lower(action_type) as action_type,
+        cast(ad_id as {{ dbt.type_bigint() }}) as ad_id,
+        date as date_day,
+        cast(coalesce(value, 0) as {{ dbt.type_float() }}) as conversion_value
+        {# 
+            Adapted from fivetran_utils.fill_pass_through_columns() macro. 
+            Ensures that downstream summations work if a connector schema is missing one of your facebook_ads__basic_ad_actions_passthrough_metrics
+        #}
+        {% if var('facebook_ads__basic_ad_actions_passthrough_metrics') %}
+            {% for field in var('facebook_ads__basic_ad_actions_passthrough_metrics') %}
+                {% if field.transform_sql %}
+                    , coalesce(cast({{ field.transform_sql }} as {{ dbt.type_float() }}), 0) as {{ field.alias if field.alias else field.name }}
+                {% else %}
+                    , coalesce(cast({{ field.alias if field.alias else field.name }} as {{ dbt.type_float() }}), 0) as {{ field.alias if field.alias else field.name }}
+                {% endif %}
+            {% endfor %}
```

Comment:
Should we consider making this a macro?

Reply:
going back and forth on this one, as the code is more complex than what's usually in our staging models, but it is really only used in this one model, so creating a macro wouldn't make us any DRYer. what are your thoughts?

Reply:
That's a good point. If it won't make things much dryer then there is no point in wrapping it up in a macro. I am good leaving as is. Thanks!

Reply:
this is now in 2 models, should i macro-fy?

Reply:
@fivetran-jamie yeah since we need to maintain it in two separate places it makes sense to macro-fy. Thanks!

Reply:
@fivetran-jamie can we macro-fy this or is it not entirely necessary?

Reply:
added!